Cannot install dependencies and run make lint in virtual environment

When installing the dependencies using Python 3.6, pip 3.6 and running "pip install --user --editable .[dev]", inside a virtual environment, dulwich package cannot be installed:
dulwich error:
creating build/temp.linux-x86_64-3.6/dulwich
x86_64-linux-gnu-gcc -pthread -DNDEBUG -g -fwrapv -O2 -Wall -Wstrict-prototypes -g -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -fPIC -I/usr/include/python3.6m -I/home/cprovencher/venv/include/python3.6m -c dulwich/_objects.c -o build/temp.linux-x86_64-3.6/dulwich/_objects.o
dulwich/_objects.c:21:20: fatal error: Python.h: No such file or directory
compilation terminated.
error: command 'x86_64-linux-gnu-gcc' failed with exit status 1

"make lint" then proceeds to make unwanted changes that result in CI failure.
"make lint" is also not currently configurable so it cannot skip directories or files such as "venv" virtual environment folder, which will also result in a make lint failure.

--- a/docs/source/contributing.rst
+++ b/docs/source/contributing.rst
@@ -11,6 +11,12 @@ Contributions to this repository must pass tests and linting.
 Install Contribution Dependencies
 ---------------------------------
 
+On Ubuntu, install system requirements:
+
+.. code:: sh
+
+   apt install -y gcc python3-dev
+
 Install dependencies in a virtual environment.
 
 .. code:: sh
@@ -34,7 +40,7 @@ and on Ubuntu with ``apt``:
 
 .. code:: sh
 
-    apt-get install -y enchant
+    apt install -y enchant
 
 Linting Bash requires `shellcheck`_:
 This can be installed on macOS, for example, with `Homebrew`_:

